src/box/lua/config/utils/schema.lua

```diff
@@ -26,10 +26,16 @@ local scalars = {}
 
 -- Create walkthrough context.
 local function walkthrough_start(self, params)
-    local ctx = {path = {}, name = rawget(self, 'name')}
-    for k, v in pairs(params or {}) do
-        ctx[k] = v
+    local ctx = table.deepcopy(params or {})
+
+    if ctx.path == nil then
+        ctx.path = {}
+    end
+
+    if ctx.name == nil then
+        ctx.name = rawget(self, 'name')
     end
+
     return ctx
 end
```

Comment:
I don't like using `deepcopy` without a good reason. If we use it everywhere we store something in a field, we perform a lot of copies of data that is never modified.

I propose to perform a shallow copy of `journey`, because we're going to modify it.

Reply:
Applied. Thank you.
Why you don't like `deepcopy`?

Reply:
> Why you don't like `deepcopy`?

Isn't it the answer?

> If we use it everywhere we store something in a field, we perform a lot of copies of data that is never modified.

I can add some thoughts here. For me it looks like a 'strong solution': if we have no good knowledge about the code, we try to leave it no chances to work in a bad way. This way it is harder to deduce assumed constraints from the resulting code, because it accepts too many broken inputs (without any failure).

If we copy `journey`, we highlight that it is modified inside. It feeds more information to a reader of the code.

----

Can we copy `journey` in the methods, where we pass it to `walkthrough_start` (`get` and `set`)? The `walkthrough_start` function shouldn't have knowledge about particular fields in the walkthrough context.

Reply:
Thank you! For some reason, I've though `walkthrough_*` uses `journey` option. Not other methods using `walkthrough_*` abstraction. 

I've applied executing a shallow copy in `schema:get()`, `schema:set()` methods

Reply:
Thanks! It seems, it should be reflected in the commit message/pull request description.

Reply:
Once updated, please, set `full-ci`. Everything else seems OK for me.

Reply:
I've rewritten the commit message if you think it needs clarification. Thank you. See PR message.